### PR TITLE
GCW-3285- Add a 'reopen' button to the offer details page

### PR DESCRIPTION
**Pull request by abulsayyad123**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-3285

### What does this PR do?

FEATURE: This PR adds feature to reopen closed/cancelled offers.

### CODE REVIEW

**Review by swatijadhav — APPROVED** [review]

**Review by shreyas098 — APPROVED** [review]

File: app/controllers/review_offer.js [hunk 1]
```diff
@@ -113,6 +113,13 @@ export default Ember.Controller.extend(AsyncTasksMixin, {
       );
     },
 
+    reopenOffer() {
+      this.runTask(() => {
+        this.get("offerService").reopenOffer(this.get("model"));
+        this.send("toggleOfferOptions");
+      });
+    },
```
Comment (patrixr) [inline app/controllers/review_offer.js:121]:
```javascript
    reopenOffer() {
      this.runTask(async () => {
        await this.get("offerService").reopenOffer(this.get("model"));
        this.send("toggleOfferOptions");
      });
    },
```

If you don't await, the async mixin will think it's over immediately and won't render the spinner correctly

**Review by patrixr — APPROVED** [review]
Just one thing to fix, you can merge right after :)

File: app/controllers/review_offer.js (hunk 1 above)
Comment (abulsayyad123) [reply app/controllers/review_offer.js:121]: sure..i'll add this change.

**Review by bharat619 — APPROVED** [review]

**Comment by abulsayyad123** [comment]
Closing this PR, it is too behind from master . Creating new branch.

**Pull request closed without merging**